app/scripts/modules/amazon/src/function/configure/Network.tsx

```diff
@@ -0,0 +1,248 @@
+import * as React from 'react';
+import { Option } from 'react-select';
+import { IPromise } from 'angular';
+import { Observable, Subject } from 'rxjs';
+import { forOwn } from 'lodash';
+
+import {
+  Application,
+  FormikFormField,
+  IWizardPageComponent,
+  IVpc,
+  ISubnet,
+  HelpField,
+  IAccount,
+  IRegion,
+  ReactSelectInput,
+  TetheredSelect,
+  SubnetReader,
+  ReactInjector,
+  ISecurityGroupsByAccountSourceData,
+} from '@spinnaker/core';
+import { FormikErrors, FormikProps } from 'formik';
+import { IAmazonFunctionUpsertCommand } from 'amazon/index';
+import { VpcReader } from 'amazon/vpc';
+
+export interface ISubnetOption {
+  subnetId: string;
+  vpcId: string;
+}
+
+export interface INetworkProps {
+  formik: FormikProps<IAmazonFunctionUpsertCommand>;
+  isNew?: boolean;
+  app: Application;
+}
+
+export interface INetworkState {
+  vpcOptions: Array<{}>;
+  accounts: IAccount[];
+  regions: IRegion[];
+  subnets: ISubnetOption[];
+  securityGroups: ISecurityGroupsByAccountSourceData;
+}
+
+export class Network extends React.Component<INetworkProps, INetworkState>
+  implements IWizardPageComponent<IAmazonFunctionUpsertCommand> {
+  constructor(props: INetworkProps) {
+    super(props);
+    this.getAllVpcs();
+  }
+
+  public state: INetworkState = {
+    vpcOptions: [],
+    accounts: null,
+    regions: [],
+    subnets: [],
+    securityGroups: null,
+  };
+  private props$ = new Subject<INetworkProps>();
+  private destroy$ = new Subject<void>();
+
+  private getAllVpcs(): void {
+    VpcReader.listVpcs().then(Vpcs => {
+      this.setState({ vpcOptions: Vpcs });
+    });
+  }
+
+  public validate(): FormikErrors<IAmazonFunctionUpsertCommand> {
+    return {};
+  }
+
+  private getAvailableSubnets(): IPromise<ISubnet[]> {
+    return SubnetReader.listSubnetsByProvider('aws');
+  }
+
+  private getAvailableSecurityGroups(): IPromise<ISecurityGroupsByAccountSourceData> {
+    return ReactInjector.securityGroupReader.getAllSecurityGroups();
+  }
+  private makeSubnetOptions(availableSubnets: ISubnet[]): ISubnetOption[] {
+    const subOptions: ISubnetOption[] = [];
+    availableSubnets.forEach(s => {
```

Comment:
simplified:
```
const subOptions: ISubnetOption[] = availableSubnets.map(s => ({ subnetId: s.id, vpcId: s.vpcId }));
```

Reply:
Done. Change will appear in the next push.